[BUG] False positive mypy error when using telegram.ext

<!--
Thanks for reporting issues of python-telegram-bot!

Use this template to notify us if you found a bug.

To make it easier for us to help you please enter detailed information below.

Please note, we only support the latest version of python-telegram-bot and
master branch. Please make sure to upgrade & recreate the issue on the latest
version prior to opening an issue.
-->
### Steps to reproduce

1. Install `mypy`
2. Write `import telegram.ext` and `var = telegram.ext.CallbackContext()`
3. Run `mypy`

### Expected behaviour

Only the error
```
Too few arguments for "CallbackContext"  [call-arg] (3:7)
```
should be thrown.

I'm pretty sure this used to be ok in a previous version.

### Actual behaviour

The errors
```
Too few arguments for "CallbackContext"  [call-arg] (3:7)
Need type annotation for 'var'  [var-annotated] (3:7)
```
are thrown.

### Configuration
**Operating System:**

macOS 11.4 (20F71)

**Version of Python, python-telegram-bot & dependencies:**

```
python-telegram-bot 13.7
Bot API 5.3
certifi 2020.12.05
Python 3.9.5 (default, May  4 2021, 03:36:27)  [Clang 12.0.0 (clang-1200.0.32.29)]
```

### Logs
Insert logs here (if necessary)

> Maybe you want to PR for this?

If I replace

```diff
-    def __init__(self, dispatcher: 'Dispatcher'):
+    def __init__(self, dispatcher: 'Dispatcher[CCT, UD, CD, BD]'):
```

the `flake8` and `mypy-ptb` pre-commit hooks fail with the errors

```
telegram/ext/callbackcontext.py:108:36: F821 undefined name 'CCT'
```

and

```
telegram/ext/callbackcontext.py:108: error: Name 'CCT' is not defined  [name-defined]
```

---

If I replace

```diff
-from telegram.ext.utils.types import UD, CD, BD
+from telegram.ext.utils.types import CCT, UD, CD, BD
```

the `pylint` pre-commit hook fails with the error

```
telegram/ext/callbackcontext.py:38:0: W0611: Unused CCT imported from telegram.ext.utils.types (unused-import)
```

So they're contradicting themselves :(

--- a/telegram/ext/callbackcontext.py
+++ b/telegram/ext/callbackcontext.py
@@ -30,7 +30,6 @@ from typing import (
     Union,
     Generic,
     Type,
-    TypeVar,
 )
 
 from telegram import Update, CallbackQuery
@@ -41,8 +40,6 @@ if TYPE_CHECKING:
     from telegram import Bot
     from telegram.ext import Dispatcher, Job, JobQueue
     from telegram.ext.utils.types import CCT
-
-CC = TypeVar('CC', bound='CallbackContext')
 
 
 class CallbackContext(Generic[UD, CD, BD]):
@@ -106,7 +103,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         '__dict__',
     )
 
-    def __init__(self: 'CallbackContext[UD, CD, BD]', dispatcher: 'Dispatcher[CCT, UD, CD, BD]'):
+    def __init__(self: 'CCT', dispatcher: 'Dispatcher[CCT, UD, CD, BD]'):
         """
         Args:
             dispatcher (:class:`telegram.ext.Dispatcher`):
@@ -226,13 +223,13 @@ class CallbackContext(Generic[UD, CD, BD]):
 
     @classmethod
     def from_error(
-        cls: Type[CC],
+        cls: Type['CCT'],
         update: object,
         error: Exception,
-        dispatcher: 'Dispatcher',
+        dispatcher: 'Dispatcher[CCT, UD, CD, BD]',
         async_args: Union[List, Tuple] = None,
         async_kwargs: Dict[str, object] = None,
-    ) -> CC:
+    ) -> 'CCT':
         """
         Constructs an instance of :class:`telegram.ext.CallbackContext` to be passed to the error
         handlers.
@@ -262,7 +259,9 @@ class CallbackContext(Generic[UD, CD, BD]):
         return self
 
     @classmethod
-    def from_update(cls: Type[CC], update: object, dispatcher: 'Dispatcher') -> CC:
+    def from_update(
+        cls: Type['CCT'], update: object, dispatcher: 'Dispatcher[CCT, UD, CD, BD]'
+    ) -> 'CCT':
         """
         Constructs an instance of :class:`telegram.ext.CallbackContext` to be passed to the
         handlers.
@@ -277,7 +276,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         Returns:
             :class:`telegram.ext.CallbackContext`
         """
-        self = cls(dispatcher)
+        self = cls(dispatcher)  # type: ignore[arg-type]
 
         if update is not None and isinstance(update, Update):
             chat = update.effective_chat
@@ -296,7 +295,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         return self
 
     @classmethod
-    def from_job(cls: Type[CC], job: 'Job', dispatcher: 'Dispatcher') -> CC:
+    def from_job(cls: Type['CCT'], job: 'Job', dispatcher: 'Dispatcher[CCT, UD, CD, BD]') -> 'CCT':
         """
         Constructs an instance of :class:`telegram.ext.CallbackContext` to be passed to a
         job callback.
@@ -311,7 +310,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         Returns:
             :class:`telegram.ext.CallbackContext`
         """
-        self = cls(dispatcher)
+        self = cls(dispatcher)  # type: ignore[arg-type]
         self.job = job
         return self
 

--- a/telegram/ext/callbackcontext.py
+++ b/telegram/ext/callbackcontext.py
@@ -40,6 +40,7 @@ from telegram.ext.utils.types import UD, CD, BD
 if TYPE_CHECKING:
     from telegram import Bot
     from telegram.ext import Dispatcher, Job, JobQueue
+    from telegram.ext.utils.types import CCT
 
 CC = TypeVar('CC', bound='CallbackContext')
 
@@ -105,7 +106,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         '__dict__',
     )
 
-    def __init__(self, dispatcher: 'Dispatcher'):
+    def __init__(self: 'CallbackContext[UD, CD, BD]', dispatcher: 'Dispatcher[CCT, UD, CD, BD]'):
         """
         Args:
             dispatcher (:class:`telegram.ext.Dispatcher`):
@@ -125,7 +126,7 @@ class CallbackContext(Generic[UD, CD, BD]):
         self.async_kwargs: Optional[Dict[str, object]] = None
 
     @property
-    def dispatcher(self) -> 'Dispatcher':
+    def dispatcher(self) -> 'Dispatcher[CCT, UD, CD, BD]':
         """:class:`telegram.ext.Dispatcher`: The dispatcher associated with this context."""
         return self._dispatcher
 

